jobs/admin.py

```diff
@@ -100,68 +101,86 @@ def not_expired(self, obj):
     not_expired.admin_order_field = 'not_expired'
 
     def assign_reviewer(self, request, id):
-        post = get_object_or_404(self.model, id=id)
-        post.assign(request.user)
-        messages.add_message(
-            request,
-            messages.INFO,
-            '{0} is now assigned.'.format(post.title)
-        )
-        return redirect('/admin/jobs/%s/%s/' % (self.get_print_name(), id))
+        try:
+            post = get_object_or_404(self.model, id=id)
+            post.assign(request.user)
+            messages.add_message(
+                request,
+                messages.INFO,
+                '{0} is now assigned.'.format(post.title)
+            )
+            return redirect('/admin/jobs/%s/%s/' % (self.get_print_name(), id))
+        except AssertionError:
+            return HttpResponseBadRequest("Post in a wrong state")
```

Comment:
Message could be more informative for people who don't know the code well - e.g. "Unable to assign reviewer; post is not in correct state"